sympy/polys/tests/test_hypothesis.py

```diff
@@ -4,40 +4,31 @@
 from sympy.polys.polytools import Poly
 
 
-@given(
-    coefficients1=st.lists(st.integers()),
-    coefficients2=st.lists(st.integers()),
-    coefficients3=st.lists(st.integers()),
-)
-def test_gcd_hypothesis(coefficients1, coefficients2, coefficients3):
-    f = Poly(coefficients1, x, domain="ZZ")
-    g = Poly(coefficients2, x, domain="ZZ")
-    r = Poly(coefficients3, x, domain="ZZ")
+def polys(*, nonzero=False, domain="ZZ"):
+    coeff_st = st.lists(st.integers())
```

Comment:
I don't think that it is generally extensible this way. 
Polys with different `domain` parameter may expect more broader than integers
I just expect things to be `st.lists(domainZZ())`

Reply:
One thing I may understand the context is that people used to test very trivial cases of `domain`, for example, `test_poly_hypothesis` used very trivial case of rationals, where rationals are just integers.

this is a valid refactory in that regards, but I think that this can misguide some people about reading the code.

Reply:
Well, it turns out to also pass with rational coefficients and domain=QQ, so I've expanded the test accordingly and checking with [HypoFuzz](https://hypofuzz.com/) shows that we plateau our coverage pretty quickly so I think it's unlikely for longer search to do much:
![image](https://github.com/sympy/sympy/assets/12229877/b77b7d71-37b7-4bcf-9f72-78ccbcd35a38)

(@asmeurer, if you're interested in using the fuzzer you can [steal Hypothesis' own dogfooding configuration](https://github.com/HypothesisWorks/hypothesis/blob/master/.github/workflows/fuzz.yml))